components/datetime/src/format/time_zone.rs

```diff
@@ -29,8 +27,36 @@ impl<'l, T> Writeable for FormattedTimeZone<'l, T>
 where
     T: TimeZoneInput,
 {
+    /// Format time zone with fallbacks.
     fn write_to<W: fmt::Write + ?Sized>(&self, sink: &mut W) -> fmt::Result {
-        write_zone(self.time_zone_format, self.time_zone, sink).map_err(|_| core::fmt::Error)
+        match self.write_no_fallback(sink) {
+            Ok(Ok(r)) => Ok(r),
+            _ => match self.time_zone_format.fallback_unit {
+                TimeZoneFormatUnit::LocalizedGmt(fallback) => {
+                    match fallback.format(
+                        sink,
+                        self.time_zone,
+                        &self.time_zone_format.data_payloads,
+                    ) {
+                        Ok(Ok(r)) => Ok(r),
+                        Ok(Err(e)) => Err(e),
+                        Err(e) => Err(e).map_err(|_| core::fmt::Error),
```

Comment:
Nit: I think it's not good style to drop errors like this. However, in this case, the function is not fallible; you need to do this because you are leveraging the `fallback_unit` enum. Could you add a `debug_assert` that this code path is not reachable?

```suggestion
                        Err(_e) => {
                            debug_assert!(false, "{:?}", _e);
                            Err(core::fmt::Error)
                        }
```

Reply:
Done.